app/src/main/java/com/keylesspalace/tusky/components/compose/ComposeViewModel.kt

```diff
@@ -325,48 +326,39 @@ class ComposeViewModel @Inject constructor(
         when (token[0]) {
             '@' -> {
-                return try {
-                    api.searchAccounts(query = token.substring(1), limit = 10)
-                        .blockingGet()
-                        .map { ComposeAutoCompleteAdapter.AccountResult(it) }
-                } catch (e: Throwable) {
-                    Log.e(TAG, String.format("Autocomplete search for %s failed.", token), e)
-                    emptyList()
-                }
+                return api.searchAccountsCall(query = token.substring(1), limit = 10)
+                    .result()
+                    .fold({ accounts ->
+                        accounts.map { AutocompleteResult.AccountResult(it) }
+                    }, { e ->
+                        Log.e(TAG, "Autocomplete search for $token failed.", e)
+                        emptyList()
+                    })
             }
             '#' -> {
-                return try {
-                    api.searchObservable(query = token, type = SearchType.Hashtag.apiParameter, limit = 10)
-                        .blockingGet()
-                        .hashtags
-                        .map { ComposeAutoCompleteAdapter.HashtagResult(it) }
-                } catch (e: Throwable) {
-                    Log.e(TAG, String.format("Autocomplete search for %s failed.", token), e)
-                    emptyList()
-                }
+                return api.searchCall(query = token, type = SearchType.Hashtag.apiParameter, limit = 10)
+                    .result()
+                    .fold({ searchResult ->
+                        searchResult.hashtags.map { AutocompleteResult.HashtagResult(it.name) }
+                    }, { e ->
+                        Log.e(TAG, "Autocomplete search for $token failed.", e)
+                        emptyList()
+                    })
             }
             ':' -> {
                 val emojiList = emoji.value ?: return emptyList()
-
-                val incomplete = token.substring(1).lowercase(Locale.ROOT)
-                val results = ArrayList<ComposeAutoCompleteAdapter.AutocompleteResult>()
-                val resultsInside = ArrayList<ComposeAutoCompleteAdapter.AutocompleteResult>()
-                for (emoji in emojiList) {
-                    val shortcode = emoji.shortcode.lowercase(Locale.ROOT)
-                    if (shortcode.startsWith(incomplete)) {
-                        results.add(ComposeAutoCompleteAdapter.EmojiResult(emoji))
-                    } else if (shortcode.indexOf(incomplete, 1) != -1) {
-                        resultsInside.add(ComposeAutoCompleteAdapter.EmojiResult(emoji))
-                    }
-                }
-                if (results.isNotEmpty() && resultsInside.isNotEmpty()) {
-                    results.add(ComposeAutoCompleteAdapter.ResultSeparator())
+                val incomplete = token.substring(1)
+
+                return emojiList.filter { emoji ->
```

Comment:
I wonder if it's the case for `sequence()`

Reply:
Should I benchmark it? My guess it that there is virtually no difference

Reply:
Just avoiding to allocate 2 lists, probably no difference.